From f43792da48073426bf5782584e3c885f781af970 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 19 Dec 2023 12:52:23 +0200
Subject: [PATCH] [SW-0] gaudi3: pif: stop monitoring time after timeout

Pif Arc is monitoring the time takes for the pci link to come up, if
timeout has occured an error bit is set. After a timeout has detected
and the error bit is set there is no point continue setting it over and
over again

Change-Id: Id0b8e7aae7ea5db171ac0a98f505e5183eeae0b5
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 88ce53f..b5aa2b9 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1052,6 +1052,8 @@
 
 void pif_arc_pcie_monitoring(void)
 {
+	bool timeout_mon = true;
+
 	/* Wait for PRSTn */
 	while (!regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PRSTN)))
 		;
@@ -1075,8 +1077,12 @@
 		if (prstn && smlh && rdlh)
 			break;
 
-		if (pif_ts_timeout_expired(0, LINK_TIMEOUT_REPORT_US))
-			hl_pif_report_err(PIF_ERR_PCI_LINK_UP_TIMEOUT);
+		if (timeout_mon) {
+			if (pif_ts_timeout_expired(0, LINK_TIMEOUT_REPORT_US)) {
+				hl_pif_report_err(PIF_ERR_PCI_LINK_UP_TIMEOUT);
+				timeout_mon = false;
+			}
+		}
 
 		/*
 		 * Some host assert prstn during link up while moving to GEN3
